src/rust/ide/src/model/suggestion_database/example.rs

```diff
@@ -0,0 +1,93 @@
+//! A module with structures related to examples found in Suggestion Database.
+use crate::prelude::*;
+
+use crate::double_representation::definition::DefinitionName;
+use crate::double_representation::module;
+use crate::double_representation::definition;
+
+use parser::Parser;
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+#[allow(missing_docs)]
+#[derive(Clone,Copy,Debug,Fail)]
+#[fail(display = "Invalid example code.")]
+pub struct InvalidExample;
+
+
+
+// ===============
+// === Example ===
+// ===============
+
+/// Example is a titled piece of code user can put into their graph to see and learn how to use
+/// our language.
+///
+/// If user picks the example, the `code` should became a body of a new method defined in current
+/// module. On scene the call for this method should appear.
+#[allow(missing_docs)]
+#[derive(Clone,Debug,Default,Eq,PartialEq)]
+pub struct Example {
+    pub name          : String,
+    pub code          : String,
+    pub documentation : String,
+}
+
+impl Example {
+    /// Return the example name converted in such way, that
```

Comment:
missing part of the sentence (if the example was meant to explain it, it is too hard to understand to me, an easy explanation should be provided here as well)